### PR TITLE
CAMEL-14576: camel-hbase - Test using test containers

### PR DESCRIPTION
**Pull request by manonthegithub**
ref #CAMEL-14576

### CODE REVIEW

**Comment by manonthegithub** [comment]
There was a dependency hell, which was making conflicts during builds.

It was not clear why you used specific version of apache hadoop, which is different to one used in the hbase-client, so I removed it. 

Please check if we can do that.

File: components/camel-hbase/pom.xml [hunk 1]
```diff
@@ -208,36 +75,18 @@
...
         </dependency>
-        <!-- because hbase-server 1.1.1 use hadoop2.5.1 by default, check is it still required by the next version update -->
         <dependency>
-            <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-hdfs</artifactId>
-            <version>${hadoop2-version}</version>
-            <classifier>tests</classifier>
+            <groupId>org.apache.zookeeper</groupId>
+            <artifactId>zookeeper</artifactId>
+            <version>3.4.6</version>
```
[the remark this answers is not in the document]
Comment (manonthegithub) [reply components/camel-hbase/pom.xml]: I used this version, cause it's used in hbase-test util, and it is not working with the version automatically taken (i suppose it its overriden by  some other dep)

File: components/camel-hbase/pom.xml [hunk 2]
```diff
@@ -208,36 +75,18 @@
...
         </dependency>
-        <!-- because hbase-server 1.1.1 use hadoop2.5.1 by default, check is it still required by the next version update -->
         <dependency>
-            <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-client</artifactId>
-            <version>${hadoop2-version}</version>
-            <classifier>tests</classifier>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>12.0.1</version>
```
[the remark this answers is not in the document]
Comment (manonthegithub) [reply components/camel-hbase/pom.xml]: I used this version, cause it's used in hbase-test util, and it is not working with the version automatically taken (i suppose it its overriden by  some other dep)

**Comment by manonthegithub** [comment]
Strange... I did the sourcecheck before committing and still got an error there :). I'll make the fixes most likely today.

File: components/camel-hbase/pom.xml [hunk 3]
```diff
@@ -208,36 +75,18 @@
...
-                    <groupId>junit</groupId>
-                    <artifactId>junit</artifactId>
-                </exclusion>
             </exclusions>
         </dependency>
-        <!-- because hbase-server 1.1.1 use hadoop2.5.1 by default, check is it still required by the next version update -->
         <dependency>
-            <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-client</artifactId>
-            <version>${hadoop2-version}</version>
```
Comment (manonthegithub) [inline components/camel-hbase/pom.xml]: I have a question, why you needed to replace the hadoop deps of the hbase client?

**Comment by davsclaus** [comment]
We are using a very old version of hbase. So it would also be good after this PR to look at upgrading it to latest 2.x version. You are welcome to create a JIRA ticket about this and help upgrade.

File: components/camel-hbase/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (davsclaus) [reply components/camel-hbase/pom.xml]: Yeah unfortunately we need to use old version for testing this old hbase version. IMHO as its a test scoped dependency then its okay to stay here as version number, and we should look at upgrading to newer hbase also.

File: components/camel-hbase/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (davsclaus) [reply components/camel-hbase/pom.xml]: Yeah unfortunately we need to use old version for testing this old hbase version. IMHO as its a test scoped dependency then its okay to stay here as version number, and we should look at upgrading to newer hbase also.

**Comment by manonthegithub** [comment]
How do you work with commits in PRs? Do you do rebase + squash before merging, so that I do not need to rebase myself, and can push new commits safely without the need to specify issue details, etc for each of them?

**Comment by davsclaus** [comment]
Yeah github allows easily to squash and merge on top of master, only when there are merge conflicts then you would need to rebase and fix.

**Comment by manonthegithub** [comment]
The sourcecheck step failed with the following error:
```
[ERROR] Failed to execute goal on project camel-hbase: Could not resolve dependencies for project org.apache.camel:camel-hbase:jar:3.2.0-SNAPSHOT: Could not find artifact jdk.tools:jdk.tools:jar:1.6 at specified path /opt/hostedtoolcache/jdk/11.0.6/x64/../lib/tools.jar -> [Help 1]
```

I don't have this error. What could be wrong?

**Comment by davsclaus** [comment]
You mean the CI server, then dont worry

**Pull request merged**